src/realm/util/file.hpp

```diff
@@ -598,6 +595,7 @@ class File {
     std::string get_path() const;
     // Return false if the file doesn't exist. Otherwise uid will be set.
     static bool get_unique_id(const std::string& path, UniqueID& uid);
+    static UniqueID get_unique_id(FileDesc file);
```

Comment:
`FileDesc` is a file descriptor/handle - can you have one if the file doesn't exist?

Reply:
Oh, I misread. I thought this was just a more convenient overload of the above that returns the  id rather than using an out param and I was assuming threw on not exists. My bad.